File: dataModule/src/main/java/com/mobnetic/coinguardian/model/market/Ripio.java
package com.mobnetic.coinguardian.model.market;

import java.util.HashMap;
import java.util.LinkedHashMap;

import org.json.JSONObject;

import com.mobnetic.coinguardian.model.CheckerInfo;
import com.mobnetic.coinguardian.model.Market;
import com.mobnetic.coinguardian.model.Ticker;
import com.mobnetic.coinguardian.model.currency.Currency;
import com.mobnetic.coinguardian.model.currency.VirtualCurrency;

public class Ripio extends Market {

	private final static String NAME = "Ripio";
	private final static String TTS_NAME = NAME;
	private final static String URL = "https://www.ripio.com/api/v1/rates/";
	private final static HashMap<String, CharSequence[]> CURRENCY_PAIRS = new LinkedHashMap<String, CharSequence[]>();
	static {
		CURRENCY_PAIRS.put(VirtualCurrency.BTC, new String[]{
				Currency.ARS
			});
	}
	
	public Ripio() {
		super(NAME, TTS_NAME, CURRENCY_PAIRS);
	}

	@Override
	public String getUrl(int requestId, CheckerInfo checkerInfo) {
		return URL;
	}
	
	@Override
	protected void parseTickerFromJsonObject(int requestId, JSONObject jsonObject, Ticker ticker, CheckerInfo checkerInfo) throws Exception {
		final JSONObject ratesJsonObject = jsonObject.getJSONObject("rates");
		ticker.bid = ratesJsonObject.getDouble("ARS_SELL");		// reversed
		ticker.ask = ratesJsonObject.getDouble("ARS_BUY");		// reversed
		ticker.last = ticker.ask;
	}
}

File: dataModule/src/main/java/com/aneonex/bitcoinchecker/datamodule/model/market/Ripio.kt
package com.aneonex.bitcoinchecker.datamodule.model.market

import com.aneonex.bitcoinchecker.datamodule.model.CheckerInfo
import com.aneonex.bitcoinchecker.datamodule.model.Market
import com.aneonex.bitcoinchecker.datamodule.model.Ticker
import com.aneonex.bitcoinchecker.datamodule.model.currency.Currency
import com.aneonex.bitcoinchecker.datamodule.model.currency.CurrencyPairsMap
import com.aneonex.bitcoinchecker.datamodule.model.currency.VirtualCurrency
import org.json.JSONObject

class Ripio : Market(NAME, TTS_NAME, CURRENCY_PAIRS) {
    companion object {
        private const val NAME = "Ripio"
        private const val TTS_NAME = NAME
        private const val URL = "https://www.ripio.com/api/v1/rates/"
        private val CURRENCY_PAIRS: CurrencyPairsMap = CurrencyPairsMap()

        init {
            CURRENCY_PAIRS[VirtualCurrency.BTC] = arrayOf(
                    Currency.ARS
            )
        }
    }

    override fun getUrl(requestId: Int, checkerInfo: CheckerInfo): String {
        return URL
    }

    @Throws(Exception::class)
    override fun parseTickerFromJsonObject(requestId: Int, jsonObject: JSONObject, ticker: Ticker, checkerInfo: CheckerInfo) {
        val ratesJsonObject = jsonObject.getJSONObject("rates")
        ticker.bid = ratesJsonObject.getDouble("ARS_SELL") // reversed
        ticker.ask = ratesJsonObject.getDouble("ARS_BUY") // reversed
        ticker.last = ticker.ask
    }
}